aspnetcore/fundamentals/routing.md

```diff
@@ -5,275 +5,386 @@ description: Discover how ASP.NET Core routing is responsible for matching HTTP
+The preceding example includes a single *route to code* endpoint using the [MapGet](xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.MapGet*) method:
 
-This basic example includes a single *Route to Code* endpoint using the [MapGet](xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.MapGet*) method. 
-* If an HTTP `GET` request is sent to the root URL `/` (`http://localhost:5000/` by default) then the request delegate shown here will execute, and `Hello World!` will be written to the HTTP resonse.
-* If the request is not a `GET` or if the URL is anything else, this will not be a match and the result will be an HTTP 404.
+* When an HTTP `GET` request is sent to the root URL `/`:
+  * The request delegate shown executes.
+  * `Hello World!` is written to the HTTP response. By default, the root URL `/` is `http://localhost:5000/`.
+* If the request is not a `GET` or if the URL is anything else, no route matches and an HTTP 404 is returned.
 
-The `MapGet` method is used to define an *endpoint* - that is, something that can be selected (by matching the URL and HTTP method), and can be executed (by running the delegate).
+### Endpoint
 
-`UseEndpoints` is where endpoints are configured that can be matched and executed by the app. For example, <xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.MapGet*>, <xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.MapPost*>, and [similar methods](xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions)can be used to wire up your own code to the routing system.
+<a name="endpoint"></a>
 
-ASP.NET Core frameworks provide extend the builder provided by `UseEndpoints`:
-- [MapRazorPages (Razor Pages)](xref:Microsoft.AspNetCore.Builder.RazorPagesEndpointRouteBuilderExtensions.MapRazorPages*)
-- [MapControllers (Controllers)](xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapControllers*)
-- [MapHub<THub> (SignalR)](xref:Microsoft.AspNetCore.SignalR.HubRouteBuilder.MapHub*) <!-- Review required -->
-- MapGrpcService<TService> (gRPC)
-<!-- TODO provide link to MapGrpcService -->
+The `MapGet` method is used to define an **endpoint**, that is, something that can be:
+
+* Selected, by matching the URL and HTTP method.
+* Executed, by running the delegate.
+
+`UseEndpoints` is where endpoints are configured that can be matched and executed by the app. For example, <xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.MapGet*>, <xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions.MapPost*>, and [similar methods](xref:Microsoft.AspNetCore.Builder.EndpointRouteBuilderExtensions) can be used to connect code to the routing system.
+The ASP.NET Core framework extends the builder provided by `UseEndpoints`:
```

Comment:
```suggestion
Additional methods can be used to connect ASP.NET Core framework features to the routing system:
```

I'm not entirely sure about the wording of "ASP.NET Core framework features" here. How would you refer to the group of MVC, RP, gRPC, etc?